### ISSUE TITLE
Unable to override accessibilityLabel for HeaderBackButton (for proper i18n)

### ISSUE THREAD

**Issue by MoOx** [react-navigation/react-navigation#6895]
**Current Behavior**

It's not possible to provide custom accessibilityLabel for HeaderBackButton.
A part is hardcoded (", back" or "Go back")

See https://github.com/react-navigation/react-navigation/blob/cdb2fed43dd33afc12e9d7a6cb801e9898c0a1fd/packages/stack/src/views/Header/HeaderBackButton.tsx#L157

**Expected Behavior**

- What do you expect should be happening?

I would expected a way to provide a custom accessibilityLabel, like for lot's of RN components.

👀 Just a new `accessibilityLabel` instead of "computing" one depending on another prop.

**How to reproduce**

Looking at the code should be clear enough. `@react-navigation/stack` 5.x are affected.

**Your Environment**

Not relevant.

**Comment by MoOx** [comment]
Thank you @satya164 !

**Comment by MoOx** [comment]
@satya164 could it be possible to have an accessibilityLabel passed via HeaderSegment like for the normal label? 
https://github.com/react-navigation/react-navigation/blob/424c9469e442d15b31c36ee68d6d98fb419ff9dc/packages/stack/src/views/Header/HeaderSegment.tsx#L299
